smarthint/actions/click.ts

```diff
@@ -0,0 +1,71 @@
+import { AppContext } from "../mod.ts";
+import { getSessionCookie } from "../utils/getSession.ts";
+import { PageType } from "../utils/typings.ts";
+
+export interface Props {
+  shippingPrice?: number;
+  shippingTime?: number;
+  /**
+   * @description Type of page you are setting up.
+   */
+  pageType: PageType;
+  date?: string;
+  elapsedTime?: number;
+  productPrice: number;
+  clickFeature: string;
+  term: string;
+  position: number;
+  productGroupID: string;
+  clickProduct: string;
+  positionRecommendation: string;
+}
+
+/**
+ * @title Smarthint Integration
+ * @description Pageview Click Event
+ */
+const action = async (
+  props: Props,
+  req: Request,
+  ctx: AppContext,
+): Promise<null> => {
+  const { recs, shcode } = ctx;
+  const {
+    shippingPrice,
+    shippingTime,
+    pageType,
+    date,
+    productPrice,
+    clickFeature,
+    term,
+    position,
+    productGroupID,
+    positionRecommendation,
+    clickProduct,
+  } = props;
+
+  const url = new URL(req.url);
+  const { anonymous, session } = getSessionCookie(req.headers);
+
+  await recs["GET /track/click"]({
+    date,
+    origin: url.origin,
+    shcode,
+    session,
+    pageType,
+    term,
+    position: String(position),
+    productId: productGroupID,
+    productPrice: String(productPrice),
+    shippingTime: String(shippingTime),
+    shippingPrice: String(shippingPrice),
+    anonymousConsumer: anonymous,
+    clickProduct,
+    clickFeature,
+    locationRecs: positionRecommendation,
+  }).then((r) => r.json());
```

Comment:
why converting to json if result is not being used? json parse would unnecessary increase CPU usage in this case